### ISSUE TITLE
Release 4.9.0 - Alpha 3 - E2E UX tests - GCP integration

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#24858]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.9.0-alpha3/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test. Also, visit the following [pre-release package guide](https://docs.google.com/document/d/1fW8koy8P8OMvcxzIOXWGDw9qmKlju-XuRHD-Bdv3uP4/edit) to understand how to modify certain links and urls for the correct testing of the development packages.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, etc) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the DevOps team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the DevOps team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Assign the issue to the corresponding team (QA if unsure), add the `Release testing` objective and `Urgent` priority. Communicate these to the team and QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues. Please answer the feedback section, this is a mandatory step.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/devel-pyserver team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the  team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Issue delivery and completion
- **Initial delivery:** The issue's assignee must complete the testing and deliver the results by Jul 24, 2024 and notify the @wazuh/devel-pyserver team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ)
- **Review:** The @wazuh/devel-pyserver team will assign a reviewer and add it to the `review_assignee` field in the  project. The reviewer must then review the test steps and results. Ensure that all iteration cycles are completed by Jul 25, 2024 date (issue must be in `Pending final review` status) and notify the QA team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ).
- **Auditor:** The QA team must audit, validate the results, and close the issue by Jul 26, 2024.

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Amazon Machine Images (AMI)](https://documentation-dev.wazuh.com/v4.9.0-alpha3/deployment-options/amazon-machine-images/amazon-machine-images.html) | - | - |
| Server | [Amazon Machine Images (AMI)](https://documentation-dev.wazuh.com/v4.9.0-alpha3/deployment-options/amazon-machine-images/amazon-machine-images.html) | - | - |
| Dashboard | [Amazon Machine Images (AMI)](https://documentation-dev.wazuh.com/v4.9.0-alpha3/deployment-options/amazon-machine-images/amazon-machine-images.html) | - | - |
| Agent | [Installing Wazuh agents](https://documentation-dev.wazuh.com/v4.9.0-alpha3/installation-guide/wazuh-agent/index.html) | - | CentOS 7 x86_64 |

## Test description

Configure the GCP integration in a Wazuh Manager and a Wazuh Agent.
Try both Pub/Sub and Storage integrations.
Follow the use cases from [blog post](https://wazuh.com/blog/cloud-security-posture-management/) section "Cloud security posture management simulation" and ensure the alerts are correctly displayed on the GCP dashboard. Make sure you follow the infrastructure and configuration details from the documentation below and not the blog one.

Documentation: https://documentation-dev.wazuh.com/v4.9.0-alpha3/cloud-security/gcp/index.html

## Known issues

- https://github.com/wazuh/wazuh-documentation/issues/4996 
- https://github.com/wazuh/wazuh-documentation/issues/5699 
- https://github.com/wazuh/wazuh-documentation/issues/5505 
- https://github.com/wazuh/wazuh/issues/15615 

## Conclusions

Summarize the errors detected (Known Issues included). Illustrate using the table below.:

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| :red_circle: | Test: Google Cloud Storage buckets integration | Service account lacked the necessary permissions | New issue: https://github.com/wazuh/wazuh/issues/24918 |
| :red_circle:  | Test: Adds rules to detect Google Cloud posture findings | Custom rules proposed don't match as expected according to the documentation	 | New issue: https://github.com/wazuh/wazuh/issues/24943 |

## Feedback

We value your feedback. Please provide insights on your testing experience.

- Was the testing guideline clear? Were there any ambiguities?
  - It became much clearer once I learned how to handle Google Cloud. It was my first time, but after that, everything looked good to me.
- Did you face any challenges not covered by the guideline?
  - I had some trouble with the dependencies required for GCP on the wazuh-agent, so I couldn't check some alerts.
- I spent too much time getting the environment ready rather than testing what was needed.
  - I realized that the credentials I received didn't allow me to create a bucket, so I had to work on an existing project.
  - Then, I had to find out how to install the GCP dependencies (Python 3.9) on CentOS 7.
  We should have this information available before starting the test.

## Reviewers validation

The criteria for completing this task are based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [x] @rauldpm
- [x] @wazuh/devel-pyserver

**Comment by lchico** [comment]
# Environment deployment

### Cloud VMs
- https://github.com/wazuh/internal-devel-requests/issues/1364

<details>
<summary> Wazuh server EC2 instance </summary>

```bash
[wazuh-user@wazuh-server ~]$  cat /etc/os-release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
SUPPORT_END="2025-06-30"
[wazuh-user@wazuh-server ~]$ uname -a
Linux wazuh-server 4.14.348-265.565.amzn2.x86_64 #1 SMP Fri Jun 28 23:44:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```
</details>

<details>
<summary> Wazuh agent EC2 instance </summary>

```bash
[centos@ip-172-31-66-73 ~]$  cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

[centos@ip-172-31-66-73 ~]$ uname -a
Linux ip-172-31-66-73.ec2.internal 3.10.0-1160.80.1.el7.x86_64 #1 SMP Tue Nov 8 15:48:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**Comment by lchico** [comment]
# Update
**Note:** With the credentials that I got, I wasn't able to create buckets
```log
Use "gcloud config set project [PROJECT_ID]" to change to a different project.
luis_chico@cloudshell:~ (e2e-wazuh-4-9-0-alpha3)$ gcloud storage buckets create gs://testing_4.9.0-alpha3
Creating gs://testing_4.9.0-alpha3/...
ERROR: (gcloud.storage.buckets.create) HTTPError 403: The billing account for the owning project is disabled in state absent. This command is authenticated as XXX.XXX@wazuh.com which is the active account specified by the [core/account] property
```
 so the solution will be continue working on an exist protect. 

I had to do the previous steps again so as resume:

```
xxxx@cloudshell:~ (wazuh-dev-258815)$ gcloud storage buckets create gs://testing_4-9-0-0-alpha3
Creating gs://testing_4-9-0-0-alpha3/...
```

**Comment by fdalmaup** [comment]
# Review

LGTM!

**Comment by rauldpm** [comment]
@lchico Do not reopen closed issues, open a new one and link the closed issues as previously reported
